addons/account/models/account_payment.py

```diff
@@ -115,7 +115,7 @@ def default_get(self, fields):
             'payment_type': total_amount > 0 and 'inbound' or 'outbound',
             'partner_id': False if multi else invoices[0].commercial_partner_id.id,
             'partner_type': False if multi else MAP_INVOICE_TYPE_PARTNER_TYPE[invoices[0].type],
-            'communication': ' '.join([ref for ref in invoices.mapped('reference') if ref]),
+            'communication': ' '.join([ref for ref in invoices.mapped('reference') if ref])[:2000],
```

Comment:
Idea : create in the sql.py file `MAX_SIZE_INDEXED_ROW = 2000`, it is more generic if the issue exist on other field.

Reply:
Thanks for the suggestion. 

It could be an idea, but so far the issue was reported in a very specific use case. At this point, I don't think a generic solution in sql.py is something we want. We'll consider if the issue is raised in several other places.